`Development`: Extend documentation for server and client guidelines to include DTOs and WebSocket subscriptions (#6987)
ls1intum/Artemis: docs/dev/guidelines/client.rst

            <!-- Do not do this -->
        </div>
    </div>

> 17. WebSocket Subscriptions

Author: I only use WebSocket now.

===========================

The client must not subscribe to more than 20 WebSocket topics simultaneously, regardless of the amount of exercises, lectures, courses, etc. there are for one particular user.

Best Practices:

1. Dynamic Subscription Handling: Subscribe to topics on an as-needed basis. Unsubscribe from topics that are no longer needed to keep the number of active subscriptions within the recommended limit.

> 2. Efficient Topic Aggregation: Where possible, use topic aggregation techniques to consolidate related data streams into a single subscription. Consequently, don't create a new topic if there already exists a suitable topic that can be reused.

Reviewer: ```suggestion
2. Efficient Topic Aggregation: Use topic aggregation techniques to consolidate related data streams into a single subscription wherever possible. Consequently, don't create a new topic if an existing one can be reused.
```
